test-data/unit/check-classes.test

```diff
@@ -4392,6 +4392,32 @@ def f(TB: Type[B]):
     reveal_type(TB)  # N: Revealed type is "Type[__main__.B]"
     reveal_type(TB.x)  # N: Revealed type is "builtins.int"
 
+[case testMetaclassAsAny]
+from typing import Any, ClassVar
+
+MyAny: Any
+class WithMeta(metaclass=MyAny):
+    x: ClassVar[int]
+
+reveal_type(WithMeta.a)  # N: Revealed type is "Any"
+reveal_type(WithMeta.m)  # N: Revealed type is "Any"
+reveal_type(WithMeta.x)  # N: Revealed type is "builtins.int"
```

Comment:
Maybe add a couple of tests to check/clarify that this only affects class objects, not instances of classes?

```suggestion
reveal_type(WithMeta.x)  # N: Revealed type is "builtins.int"
reveal_type(WithMeta().a)  # E: "WithMeta" has no attribute "a" \
                           # N: Revealed type is "Any"
reveal_type(WithMeta().x)  # N: Revealed type is "builtins.int"
```

Reply:
Sure!